### DISCUSSION TITLE
[FEAT] Winget Package Count Support for Windows

### DISCUSSION THREAD

**Discussion by Ash-Olorenshaw** [fastfetch-cli/fastfetch#1158] [Q&A] [answered]
Be sure to read [FAQ](https://github.com/fastfetch-cli/fastfetch?tab=readme-ov-file#faq) before submitting a new issue.

## Wanted features:
I know it'll probably come off as a bit of a joke to use `fastfetch` on Windows, but I did notice that the `Packages` section lists Scoop and Choco packages, but not Winget ones. Could we get support for the number of packages installed via Winget?

## Motivation:

I know I can just add a command module with the PowerShell command:
```powershell
((winget list) -match ' winget$').Count
```
to list the number of packages, but I thought that this would look better (and also make more sense) to be integrated in with the `Packages:` count.
No worries if the answer is no, by the way, just curious as to whether it had been considered.

**Comment by CarterLi** [comment, marked as answer]
It's disabled by default as winget package counting is extremely slow

Try `fastfetch -s packages --packages-disabled '' --processing-timeout 5000`

**Reply by Ash-Olorenshaw** [reply]
Just checked and this works perfectly, thanks!

Is there somewhere I can pass `--processing-timeout 5000` into my `config.jsonc` so that I don't have to use that flag everytime?

**Reply by CarterLi** [reply]
```jsonc
{
  "general": {
    "processingTimeout": <whatever in ms>
  }
}
```

**Comment by CarterLi** [comment]
> I know it'll probably come off as a bit of a joke to use fastfetch on Windows

Why is it joke? Fastfetch has first-class support for Windows. I developed it and I'm still working on it. If using fastfetch on Windows is a joke, my work is a joke.

I use Windows. I'm proud to use Windows.

**Reply by Ash-Olorenshaw** [reply]
> > I know it'll probably come off as a bit of a joke to use fastfetch on Windows
> 
> Why is it joke? Fastfetch has first-class support for Windows. I developed it and I'm still working on it. If using fastfetch on Windows is a joke, my work is a joke.
> 
> I use Windows. I'm proud to use Windows.

I'm sorry if it came off as offensive, posting issues on other Github pages relating to Windows has often led to people asking me if I'm serious and such because "no _real_ dev uses Windows". So for this reason I usually lead with that kind of line to just to dispel the irrelevant comments straight away.

> It's disabled by default as winget package counting is extremely slow
> 
> Try `fastfetch -s packages --packages-disabled '' --processing-timeout 5000`

Thanks, I'll try that now! I appreciate the quick response :)

**Reply by CarterLi** [reply]
Using Windows is not a joke. Ignoring Windows has 70%+ market share is a joke.

**Reply by Ash-Olorenshaw** [reply]
That's very true, it's why I always try to have feature parity as best I can between the platforms.
